### DISCUSSION TITLE
Tab-only mode does not show time duration for whole note

### DISCUSSION THREAD

**Discussion by lannka** [CoderLine/alphaTab#601] [General]
This what we have in AlphaTab right now:
<img width="404" alt="Screen Shot 2021-05-15 at 11 16 22 PM" src="https://user-images.githubusercontent.com/8496897/118387499-af608f80-b5d3-11eb-96c2-8896fc223d38.png">

I would expect something like the following:

<img width="355" alt="Screen Shot 2021-05-15 at 11 18 09 PM" src="https://user-images.githubusercontent.com/8496897/118387537-dd45d400-b5d3-11eb-90f8-ecd8bb4e5796.png">

What do you think?

**Comment by Danielku15** [comment]
alphaTab automatically activates the display of rests in the tab staff if you have the `staveProfile` set to `tab`. Can you provide more details on how exactly your setup looks like? 

![image](https://user-images.githubusercontent.com/674916/118393797-30a23b80-b641-11eb-9ef1-c807bd85f5bc.png)

**Reply by lannka** [reply]
Thanks for the response. I was referring to the regular notes on the tab, not rests.

With all notes of different durations put together, here is a cleaner example to explain the ambiguity I was referring to. You can see that the quarter notes (the last 2 notes in the first bar) and the half notes (the 2 notes in the 2nd bar) are displayed the same.

<img width="601" alt="Screen Shot 2021-05-16 at 9 13 05 AM" src="https://user-images.githubusercontent.com/8496897/118404245-f461e180-b626-11eb-88bc-dff8efae8a96.png">

```
0.1.16*4 0.2.8*2 0.3.4*2 |
0.4.2*2 |
0.4.1 |
```

----
One solution that I saw often is to use those **rests-like bold dashes** (they are thiner than rests) to extend a note. In the following example, the quarter note is extended to a full note by appending 3 bold dashes. 
<img width="355" alt="Screen Shot 2021-05-15 at 11 18 09 PM" src="https://user-images.githubusercontent.com/8496897/118404359-76eaa100-b627-11eb-9e5d-1c9cca34e60c.png">

Not sure if the notation is a standard (does tab even have a standard? :-)). But I found it super useful. Or do you know any other notations are used to make this clear?


## My setting
```js
{
                notation: {
                    rhythmMode: 'showwithbars',
                },
                display: {
                    staveProfile: 'tab',
                },
            }
```

Here is a live demo: https://alphatab-whole-note.glitch.me/

**Reply by Danielku15** [reply]
I think I got it. The issue is: the rhythm notation of quarter notes and full notes are displayed the same. 

Guitar Pro solves this by drawing a slighly shorter rhythm line and additionally an ellipse around the tab numbers: 
![image](https://user-images.githubusercontent.com/674916/118409087-12602e00-b689-11eb-9d90-fc9a3fdd02b9.png)

In fact there are a bunch of settings in Guitar Pro controlling the display of the rhythm: 
![image](https://user-images.githubusercontent.com/674916/118409457-d6c66380-b68a-11eb-9857-34ed9c4dec8a.png)

Soundslice shows a slighly shorter rhythm line for half notes: 
![image](https://user-images.githubusercontent.com/674916/118409244-b649d980-b689-11eb-8aab-7eecef5dd38f.png)

Flat.io also has a shorter rhythm line: 
![image](https://user-images.githubusercontent.com/674916/118409344-4d169600-b68a-11eb-87fe-658ba0b2273a.png)

Songsterr as well the same: 
![image](https://user-images.githubusercontent.com/674916/118409677-f447fd00-b68b-11eb-8ef6-086c19662bed.png)

MuseScore the same: 
![image](https://user-images.githubusercontent.com/674916/118409416-9830a900-b68a-11eb-905b-04e6ea43c44d.png)

You proposal I have not seen in any software before (maybe not dedicated and not in this context?). It rather seems to me that the dashes are a mechanism to indicate tied notes if there is not standard notation. By making two quarter notes tied, you also get your half note and solve the ambiguity in the display. 

That said: What do you think about a fix where I draw the rhythm line of the whole notes shorter? The circle seems rather special to Guitar Pro 7 and I would leave it out for now to be more aligned with the others.

**Reply by lannka** [reply]
Thanks for the investigation. Now I think the dashes in my examples might be borrowed from numbered notation. 

I like the idea of shortening the stem to indicate longer duration. How would you differentiate half notes and whole notes? How are whole notes rendered in those software?

**Reply by lannka** [reply]
I checked guitar pro , seems whole note does not have stem. Half note has half lengh stem.

I think we can follow this practice in AlphaTab?

![1621301648601](https://user-images.githubusercontent.com/8496897/118577018-7f81cb00-b73e-11eb-9dae-380221e948e2.jpg)

See the last 2 notes as a comparison

**Reply by Danielku15** [reply]
Added the new rendering. You can check out alphaTab 1.3.0-alpha.120.
![image](https://user-images.githubusercontent.com/674916/119239141-7716ff00-bb47-11eb-9624-24eefdad6bc4.png)

**Reply by lannka** [reply]
will try, thanks!
